### PR TITLE
arrays-programs_involving_data_sequences-easy-q6

### PR DESCRIPTION
**Pull request by SamamthaH**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [x] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/arrays/programs_involving_data_sequences/easy/q6/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+public class MainTest extends BaseTest {
+    // Parsons
+
+    //vary this variable to vary number of bmis calculated. Max of 20 (inclusive) because of default input.
+    final static int size = 10;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "180 65 135 45 187.3 95 67 23 132 78.3 184.78 67.521 704 15.4 63.5 38 374 23.4 437 38 180 65 135 45 187.3 95 67 23 132 78.3 184.78 67.521 704 15.4 63.5 38 374 23.4 437 38";
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_data_sequences/easy/q6/MainTest.java:28]: We should be able to just generate this default input with arrayUtil and then we don't need this limit on the variation,

Comment (Hedgemon4) [reply src/arrays/programs_involving_data_sequences/easy/q6/MainTest.java:28]: As it doesn't really matter what the default is

File: src/arrays/programs_involving_data_sequences/easy/q6/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        StringBuilder s = new StringBuilder();
+        for (int x = 0; x < size * 2; x++) {
+            s.append(values[x]);
+            s.append(" ");
+        }
+
+        runWithInput(s.toString());
+
+        for (int x = 0; x < size; x++) {
+            assertEquals(bmis[x], Double.parseDouble(getItemByName(String.valueOf(x + 1))), 0.00000001, "Your program calculate BMI " + (x + 1) + " correctly.");
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_data_sequences/easy/q6/MainTest.java:89]: Generally, I don't think we need to tell them at what specific line their code messed up

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/programs_involving_data_sequences/easy/q6/BMWhy.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    BMI 9: 101.65289256198345
+    BMI 10: 9.192760700948004E-4
+ */
+
+public class BMWhy {
+    public static double calculateBMI(double height, double weight) {
+        return weight / (height * height);
+    }
+
+    public static void main(String[] args) {
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_data_sequences/easy/q6/BMWhy.java:28]: Similar thing here with this question. We should change the question so that we know they are actually using an array.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]
